network/network.go

```diff
@@ -256,30 +259,34 @@ func NewNetwork(
 		ipTracker.ManuallyTrack(nodeID)
 	}
 
+	// ingressConnCount tracks connections to all remote peers,
+	// so for safety we set it as pointer, in case the config object is copied.
```

Comment:
> for safety we set it as pointer, in case the config object is copied.

The config already includes atomic values (so it already isn't safe to copy). Also, `atomic.Uint64` includes `noCopy` - which means that CI will fail if it is ever copied... Seems like it would be cleaner to just put it in the config

Reply:
I did it because it's less work for me to change the tests :) 

```
network/peer/peer_test.go:180:32: copylocks: call of newRawTestPeer copies lock value: github.com/ava-labs/avalanchego/network/peer.Config contains sync/atomic.Uint64 contains sync/atomic.noCopy (govet)
```

Reply:
Also I don't want to rely on CI for something like this...

Reply:
[Made](https://github.com/ava-labs/avalanchego/pull/3719/commits/1cf5056e7b6460f0127492102254d8fb5e9ae0b0#diff-635834cc4e56923e40bf037154f3c89d2d64e8cafe4286e3f086356b5eb2525fR29-R32) it a non pointer